python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -1867,13 +1881,46 @@ def do_evaluate(
             table = plc.copying.gather(left.table, lg, left_policy)
             result = DataFrame.from_table(table, left.column_names, left.dtypes)
         else:
+            swapped = False
             if how == "Right":
                 # Right join is a left join with the tables swapped
                 left, right = right, left
                 left_on, right_on = right_on, left_on
+                swapped = True
+            if swapped:
+                effective_order = {
+                    "none": "none",
+                    "left": "right",
+                    "right": "left",
+                    "left_right": "right_left",
+                    "right_left": "left_right",
+                }.get(maintain_order, "none")
```

Comment:
Two notes:
1. If we really want to reduce overheads we could store this dict as a constant on the class and then query it upon use.
2. Do we have to use `get` here? If so, why isn't the list of options here exhaustive?